modules/templates-writing-parameters.adoc

```diff
@@ -63,13 +63,19 @@ characters long consisting of all upper and lowercase alphabet letters
 and numbers.
 
 The syntax available is not a full regular expression syntax. However, you can
-use `\w`, `\d`, and `\a` modifiers:
+use `\w`, `\d`, `\a`, and `\A` modifiers:
 
 - `[\w]{10}` produces 10 alphabet characters, numbers, and underscores. This
 follows the PCRE standard and is equal to `[a-zA-Z0-9_]{10}`.
 - `[\d]{10}` produces 10 numbers. This is equal to `[0-9]{10}`.
 - `[\a]{10}` produces 10 alphabetical characters. This is equal to
 `[a-zA-Z]{10}`.
+- `[\A]{10}` produces 10 punctuation or symbol characters. This is equal to ``[~!@#$%\^&*()\-_+={}\[\]\\|<,>.?/"';:` ]{10}``.
```

Comment:
Is it possible to lose the space character at the end of the bracketed character class? GitHub manages to render this OK: 
``[~!@#$%\^&*()\-_+={}\[\]\\|<,>.?/"';:`]{10}``

Reply:
Or (taking into my suggestion to not bother explaining `\A` in terms of a PCRE, simply:

> `\[A]{10}` produces 10 symbol characters chosen from the following set: ``~!@#$%^&*()-_+={}[]\|<,>.?/"';:` ``

And drop the `[NOTE]` section entirely.

Reply:
@lbarbeevargas - I'll differ to your decision on this. Is the note necessary?

Reply:
I think it's more clear to have the bullet for `[\A]{10}` follow Sam's last comment:

`[\A]{10}` produces 10 punctuation or symbol characters. This is equal to ``[~!@#$%^&*()-_+={}[]\|<,>.?/"';:`]{10}``.

Therefore, the example does not include the `\` to escape specific characters. If the note about escaping `\` is useful, you can drop the `+` before the note so that it applies to all of the bullets/examples:

> Depending on if the template is written in YAML or JSON, and the type of string that the modifier is embedded within, you might need to escape the backslash with a second backslash. The following examples are equivalent:
> .Example YAML template with a modifier
> .Example JSON template with a modifier

Reply:
It's not equivalent to _that_ PCRE. Putting a `-` character inside the bracketed character class unintentionally creates a [character range](https://perldoc.perl.org/perlrecharclass#Character-Ranges). In this case, the range matches from `)` to `_` which includes digits and uppercase letters... escaping it with a `\` or moving it to the start or end fixes things.

I think the `]` has to be escaped too, otherwise doesn't it end the bracketed character class?

And the lone `\` should be escaped too.

The [perl docs cite this](https://perldoc.perl.org/perlrecharclass#%5B5%5D) as being equivalent to `[[:punct:]]`: ``[-!"#$%&'()*+,./:;<=>?@[\\\]^_`{|}~]``

Reply:
Cool, thanks for the catch! Here's the new PR with those fixes--https://github.com/openshift/openshift-docs/pull/31431. If you can review, we'll merge it.